src/configure/extract_from_c.ml

```diff
@@ -0,0 +1,118 @@
+(*
+ * Copyright (c) 2016 whitequark.
+ *
+ * This file is distributed under the terms of the MIT License.
+ * See the file LICENSE for details.
+ *)
+
+let getenv ~default name =
+  try Sys.getenv name
+  with Not_found -> default
+
+let nsplit sep str =
```

Comment:
Writing the code is a sunk cost, but maintaining it isn't.  I'd prefer to maintain a one-liner.

And yes, Str is badly designed.  Its only real virtue is that it's in the standard library.